### PR TITLE
feat: UI improvement for integration test runner

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Improvements:
- Able to run under project root dir with `cargo run --bin sqlness-runner`
- Force to run with the newest DB build.
- Hook build output. Redirect server log to file.
- No orphan process will leave.

## Checklist

- [x]  I have written the necessary rustdoc comments.
- []  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: tests/runner/src/util.rs [hunk 1]
```diff
@@ -95,3 +105,65 @@ pub fn values_to_string(data_type: ColumnDataType, values: Values) -> Vec<String
...
+    let mut workspace_root = PathBuf::from(get_workspace_root());
+
+    // change directory to target dir (workspace/target/<build mode>/)
+    workspace_root.push("target");
+    workspace_root.push(mode);
+
+    workspace_root.into_os_string().into_string().unwrap()
+}
+
+/// Spin-waiting a socket address is avaliable, or timeout.
```
Comment (dongxuwang) [inline tests/runner/src/util.rs:145]:
```suggestion
/// Spin-waiting a socket address is available, or timeout.
```

File: tests/runner/src/util.rs [hunk 2]
```diff
@@ -95,3 +105,65 @@ pub fn values_to_string(data_type: ColumnDataType, values: Values) -> Vec<String
...
+                Err(_) => time::sleep(PORT_CHECK_INTERVAL).await,
+            }
+        }
+    };
+
+    tokio::select! {
+        _ = check_task => {
+            true
+        },
+        _ = time::sleep(timeout) => {
```
Comment (evenyag) [inline tests/runner/src/util.rs:165]: Looks like you could use tokio's [`timeout`](https://docs.rs/tokio/latest/tokio/time/fn.timeout.html) function.

Comment (waynexia) [reply tests/runner/src/util.rs:165]: 👍

**Review by evenyag — APPROVED** [review]
LGTM

**Review by v0y4g3r — APPROVED** [review]
L6TM

**Pull request merged**